core/src/edu/rice/comp413fall2020grey/ObjStorage/GameObjectMetadata.java

```diff
@@ -0,0 +1,20 @@
+package edu.rice.comp413fall2020grey.ObjStorage;
+
+import java.util.UUID;
+
+/**
+ * Metadata for this object that will be stored on the registrar and used by object location.
+ */
+public interface GameObjectMetadata {
+
+  /**
+   * Unique identifier for this game object across the entire distributed system.
+   */
+  UUID getUUID();
+
+  /**
+   * Returns the type of the game object this metadata is for. This will help determine if this object is interesting to other objects.
+   */
+  Class<?> getUnderlyingType();
```

Comment:
Im a little confused how this will help with the metadata by knowing the underlying type because the registrar should have no knowledge of the underlying type

Reply:
I added this to help with the `isInterestedIn` function implementations. For example, if my game has coins, rocks, and players, the implementation of coin's interest function may only care about players, and not about rocks or other coins. This function would give a way to clearly identify the type of the object the metadata belongs to, otherwise there is no way to differentiate between a player and a rock.